vaultkeeper/vaultkeeper.py

```diff
@@ -117,26 +118,32 @@ def renew_all(self):
             if entry.renewable:
                 self.renew_lease(entry)
 
-    def run(self):
+    def cleanup(self):
+        self.vault_client.revoke_self_token()
+
+    def start_subprocess(self):
```

Comment:
We still want to call `.run()` from the outside, because conceptually (and temporally), the two operations are tightly coupled. We don't want to start the subprocess, go make a cup of coffee, and then start watching it for renewal -- the token could expire before we get there.

Instead, maybe call these `_start_subprocess` and `_watch_and_renew` (because their separation is an implementation detail that makes testing easier) and call them from `.run()`?